crates/bitwarden-ipc/src/manager.rs

```diff
@@ -0,0 +1,142 @@
+use crate::{
+    error::{ReceiveError, SendError},
+    message::Message,
+    providers::{CommunicationProvider, CryptoProvider, SessionProvider},
+};
+
+pub struct Manager<Crypto, Com, Ses>
```

Comment:
`Manager` currently deals with binary data only, but for real world use-cases we might want to add something on top of the manager which is able to serialize/de-serialize. This additional abstraction might also be a good place to implement request/response functionality